File: calc_dup/calc_map_rate.pl
#!/bio/bin/perl -w

# #Target Bases
# TruSeq : 62085295
# Sure Select 50Mb : 51860012
# Sure Select 4 + UTR : 107839035
# Sure Select 5 + UTR : 116898557
# Genome : 3095677412

my $target_bases = 3095677412;

my $name;
my $n_reads = 0;
my $map_reads = 0;
my $unq_reads = 0;
my $map_bases = 0;
my $unq_bases = 0;

while(<STDIN>){
	chomp;
	my @column = split(/\s/, $_);
	$name = $column[0];
	$n_reads += $column[1];
	$map_reads += $column[2];
	$unq_reads += $column[3];
	$map_bases += $column[4];
	$unq_bases += $column[5];
}

@path = split(/\//, $name);
@sample = split(/\./, $path[2]);

if($sample[0] =~ /SS50Mb/){
	$target_bases = 51860012;
}
elsif($sample[0] =~ /SS4UTR/){
	$target_bases = 107839035;
}
elsif($sample[0] =~ /SS5UTR/){
	$target_bases = 116898557;
}
elsif($sample[0] =~ /SS6UTR/){
	$target_bases = 90697072;
}
elsif($sample[0] =~ /TruSeq/){
	$target_bases = 62085295;
}
elsif($sample[0] =~ /WG/){
	$target_bases = 3095677412;
}
elsif($sample[0] =~ /Cancer/){
	$target_bases = 8643867;
}

#Sample,#Reads,#Mapped Reads,#Mappe Reads (Unique),Mapping Rate (%),Mapping Rate (Unique) (%),Coverage (x),Coverage (Unique) (x)
print "$sample[0],$n_reads,$map_reads,$unq_reads," . $map_reads / $n_reads * 100 . "," . $unq_reads / $n_reads * 100 . "," . $map_bases / $target_bases . "," . $unq_bases / $target_bases . "\n";

File: calc_dup/calc_map_rate.rb
#!/bio/bin/ruby

require 'yaml'
require_relative '../app/config.rb'
require_relative '../app/prepkit.rb'

module CalcMapRate

  PREP = Prepkit.new( load_config().prepkit_info )
  def self.run( sample )
    input = Dir["#{sample}/stat/map/*.Nmap"].map{ |f| File.read(f) }.join()
    return run_with_input( input )
  end
  
  def self.run_with_input( input = STDIN.read )

    name = nil
    n_reads = map_reads = unq_reads = map_bases = unq_bases = 0
    
    for l in input.split( "\n" )
      c = l.chomp.split(/\s/)
      name = c[0]
      n_reads += c[1].to_i
      map_reads += c[2].to_i
      unq_reads += c[3].to_i
      map_bases += c[4].to_i
      unq_bases += c[5].to_i

    end

    path = name.split(/\//)
    sample = path[2].split(/\./)

    target_bases = PREP.suffix2targetbases( sample[0] ) || 3095677412

    # Sample,#Reads,#Mapped Reads,#Mappe Reads (Unique),Mapping Rate (%),Mapping Rate (Unique) (%),Coverage (x),Coverage (Unique) (x)
    return "#{sample[0]},#{n_reads},#{map_reads},#{unq_reads},#{ map_reads.to_f / n_reads * 100 },#{ unq_reads.to_f / n_reads * 100 },#{ map_bases.to_f / target_bases },#{ unq_bases.to_f / target_bases }"

  end
end

if __FILE__ == $0
  if ARGV.size >= 1
    sample = ARGV[0]
    puts CalcMapRate( sample ) 
  else
    input = STDIN.read
    puts CalcMapRate.run_with_input( input ) 
  end
end
